### DISCUSSION TITLE
Is it still relevant to distribute Sozi for 32-bit architectures?

### DISCUSSION THREAD

**Discussion by aumouvantsillage** [senshu/Sozi#521] [General]
Sozi releases are usually built for Linux, Windows and MacOS X, and for 32-bit and 64-bit Intel/AMD architectures when possible.
Based on the download statistics of the latest stable release, 17% of Linux and Windows users downloaded the 32-bit package.
I see two possible reasons for that:

* Either these users actually use 32-bit computers, or run 32-bit operating systems.
* Or they had no clear idea which package was appropriate for their computers. Maybe some of them downloaded both the 32-bit and 64-bit packages, just in case.

Building and uploading packages for all platforms and architectures take time. Since there is a demand for additional package formats (RPM or flatpak for Linux) and additional architectures (ARM essentially), I am planning to drop 32-bit support in the next releases.

What do you think? Is is still relevant to ship Sozi packages for 32-bit architectures?

For background information, the Sozi presentation editor is based on [Electron](https://www.electronjs.org), a desktop application development framework using web technologies. While the Electron blog [announced that 32-bit Linux support would be discontinued](https://www.electronjs.org/blog/linux-32bit-support) starting from Electron 4, it seems that I can still build 32-bit versions of Sozi with Electron 9.

**Comment by NedNet29** [comment]
Hi, 
I feel kind of ripped in half trying to answer this question : I personally use Sozi with Windows 10, Mint and soon Fedora (my main station OS) since RPMs are becoming available (thanks a lot by the way), so I wouldn't bother seeing 32-bit disappear, all the OSs I use being 64-bit.
But for the sake of good sense trying to keep old functionnal PCs usefull, I think 32 bit should be maintained as long as it can reasonably be.
Since I can't help building releases, that's just a personal opinion based on an idealistic view, and I would totally understand you drop 32-bit if it's too time consuming.
HTH

**Comment by hoshi411** [comment]
A lot of chromebooks run crouton (linux) in 32bit environment. So the millions of Sozi users working on chromebooks would be affected.  So the question is, do you want to support chromebook users for another 2 years or so .

**Reply by aumouvantsillage** [reply]
> So the millions of Sozi users working on chromebooks

Thanks for your feedback, but I doubt there are millions of Sozi users :-)

The last time I provided 32-bit packages was in february 2021.
They represent 20% of the downloads and most of them were for WIndows.
Since then, I did not receive any request to ship 32-bit packages again.
